bin/problem.py

```diff
@@ -629,67 +646,117 @@ def testcases(
-            interaction_paths = []
+            # resolve some inconsistencies
+            if ".in" not in in_found:
+                if ".ans" in ans_found:
+                    warn(f"Found {name}.ans but no {name}.in. IGNORED.")
+                    ans_found.remove(".ans")
+                if ".out" in ans_found:
+                    warn(f"Found {name}.out but no {name}.in. IGNORED.")
+                    ans_found.remove(".out")
+            if has_statement and ".out" in ans_found:
+                # we prefer .statement files
+                warn(f"Found {name}.out (but also .statement). IGNORED.")
+                ans_found.remove(".out")
+
+            # .interaction files get highest priority
+            if return_interaction_file and name.with_suffix(".interaction").is_file():
+                if not p.interactive and not p.multi_pass:
+                    warn(f"Found {name}.interaction for  non-interactive/non-multi-pass. IGNORED.")
+                else:
+                    if has_statement:
+                        warn(
+                            f"Mixed .interaction and .statement file for {name}. (using .interaction)."
+                        )
+                    if ".out" in ans_found:
+                        warn(f"Mixed .interaction and .out file for {name}. (using .interaction).")
+                    testcases.append(name.with_suffix(".interaction"))
+                    continue
 
-        testcases = list[Path | tuple[Path, Path]]()
-        for in_path in in_paths:
-            ans_path = in_path.with_suffix(".ans")
-            if not ans_path.is_file():
-                warn(f"Found input file {in_path} without a .ans file. Skipping.")
+            if not in_found or not ans_found:
+                warn(f"Could not find valid .in/.ans combination for test case {name}. SKIPPED.")
+                warn("Numbering for statement and download could be inconsistent!")
                 continue
-            testcases.append((in_path, ans_path))
 
-        for in_path in statement_in_paths:
-            # first remove .statement, then replace .in with .ans.statement
-            ans_path = in_path.with_suffix("").with_suffix(".ans.statement")
-            if not ans_path.is_file():
-                warn(f"Found input file {in_path} without a .ans.statement file. Skipping.")
-                continue
-            testcases.append((in_path, ans_path))
+            if in_found[0] == ".in" and ans_found[0] == ".ans":
+                has_raw = True
 
-        for interaction_path in interaction_paths:
-            testcases.append(interaction_path)
+            # fall back is pair of files
+            testcases.append((name.with_suffix(in_found[0]), name.with_suffix(ans_found[0])))
 
-        testcases.sort()
+        if has_raw and not p.settings.ans_is_output:
+            warn(
+                "It is advised to overwrite .in and .ans for samples if .ans does not represent a valid output"
+            )
```

Comment:
What should a problem author do when they see this warning? Does "overwrite" mean that they should add `.*.statement` and/or `.*.download` files? If so, the warning should probably say that, to help people with less experience :slightly_smiling_face: 

Also, I receive this warning when I `bt upgrade` NWERC 2024 Hash Collision and run `bt zip`. But the `.ans` file is empty there, so I shouldn't have to overwrite anything, right? :thinking: 

The samples of this problem have valid `.in` files according to the input validator, but that's not what this warning is about. Although `has_raw` _is_ set to `True` when `.in` exists for a sample. Is that intended?

Reply:
maybe the warning is nicer now?